online-shop/src/Sagas/saga.tsx

```diff
@@ -0,0 +1,217 @@
+import { takeLatest, put, all } from 'redux-saga/effects';
+import * as PLTypes from '../ReduxStore/ProductListSection/types';
+import { IProduct } from '../Models/Models';
+import { loadProducts } from '../ReduxStore/ProductListSection/actions';
+import * as PDTypes from '../ReduxStore/ProductDetailsSection/types';
+import { loadProductDetails, activateModalRedirectToProducts, deleteProductFromStore } from '../ReduxStore/ProductDetailsSection/actions';
+import * as NPTypes from '../ReduxStore/NewProductSection/types';
+import { createNewProduct } from '../ReduxStore/NewProductSection/actions';
+import * as EPTypes from '../ReduxStore/EditProductPageSection/types';
+import { loadProductData, editProduct } from '../ReduxStore/EditProductPageSection/actions';
+import * as SCTypes from '../ReduxStore/ShoppingCartSection/types';
+import { checkoutShoppingCart, decreaseProductQuantity } from '../ReduxStore/ShoppingCartSection/actions';
+
+function* performProductListLoad() {
```

Comment:
In my view, you could improve the structure of this file by creating some utility functions for doing the fetch calls (instead of always repeating the headers, the basic content handling, etc). I would also consider splitting this in multiple files.